src/test/java/com/fasterxml/jackson/databind/deser/TestCustomSubTypeValidator.java

```diff
@@ -0,0 +1,277 @@
+            + "    },"
+            + "    {"
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Penguin\","
+            + "      \"name\" : \"Tux\""
+            + "    }"
+            + "  ]"
+            + "}";
+
+    protected static final String PETTING_ZOO_JSON = "{"
+            + "  \"name\" : \"Jacksonville Petting Zoo\","
+            + "  \"animals\": ["
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Goat\","
+            + "      \"name\" : \"Bill\""
+            + "    },"
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Sheep\","
+            + "      \"name\" : \"Shaun\""
+            + "    }"
+            + "  ]"
+            + "}";
+
+    /*
+    /**********************************************************
+    /* Helper classes (beans)
+    /**********************************************************
+     */
+
+    public static class Zoo
+    {
+        private String _name;
+
+        private List<Animal> _animals;
+
+        public String getName()
+        {
+            return _name;
+        }
+
+        public void setName(String name)
+        {
+            _name = name;
+        }
+
+        public List<Animal> getAnimals()
+        {
+            return _animals;
+        }
+
+        public void setAnimals(List<Animal> animals)
+        {
+            _animals = animals;
+        }
+    }
+
+    @JsonTypeInfo(
+        use=JsonTypeInfo.Id.CLASS,
+        include=JsonTypeInfo.As.PROPERTY,
+        property="species"
+    )
+    public static abstract class Animal
+    {
+        private String _name;
+
+        public String getName() {
+            return _name;
+        }
+
+        public void setName(String name) {
+            _name = name;
+        }
+    }
+
+    public static class Lion extends Animal
+    {
+    }
+
+    public static class Warthog extends Animal
+    {
+    }
+
+    public static class Goat extends Animal
+    {
+    }
+
+    public static class Sheep extends Animal
+    {
+    }
+
+    public static class Penguin extends Animal
+    {
+    }
+
+    /*
+    /**********************************************************
+    /* Helper classes (validators)
+    /**********************************************************
+     */
+
+    static class DenyAllValidator
+        extends SubTypeValidator
+    {
+        @Override
+        public void validateSubType(DeserializationContext ctxt, JavaType type, BeanDescription beanDesc)
+            throws JsonMappingException 
+        {
+            ctxt.reportBadTypeDefinition(
+                    beanDesc,
+                    "Illegal type (%s) to deserialize: this validator does not allow deserialization at all!",
+                    type.getRawClass().getName());
+        }
+    }
+
+    static class FriendlyAnimalValidator
```

Comment:
This validator combines the black-list approach from the default `SubTypeValidator` with a white-list approach for one specific part of the type hierarchy.

Ultimately, it is the goal of #2208 to make the black-list from `SubTypeValidator` extensible. This is not demonstrated here. But I could add a suitable `SubTypeValidator` sub-class to this PR, if desired? Or, leave that exercise to the users of Jackson databind?

Reply:
Fyi, I've added separate class `ExtensibleSubTypeValidator` to this PR now, which addresses extensibility of the default black-list. Tests for that are below...